### PR TITLE
CI: group tests in stages

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- Group tests in stages to reduce the CI pressure. In cases where a stage is failing it makes no sense to run all others.
- Additionally, the second commit set the environment variable `ALPAKA_CI` to avoid running the `headerCheck` if not explicitly activated.
- Disable `randomCells2D` for the OpenACC backend because of invalid memory access. The reason for the issue should be investigated independently to avoid blocking other PRs.

Per YAML file I incremented the stage id, I have not balanced the stages by hand because this would make it hard to keep an overview.

Writing everything by hand should be only a workaround we need asap a generator https://github.com/alpaka-group/alpaka/issues/1517#issuecomment-988044352

### CODE REVIEW

**Comment by bernhardmgruber** [comment]
I see `nvhpc` in the last 8th stage of the pipeline and wonder whether we should have it earlier to avoid running 7 stages to then fail in the 8th because of a very unique compiler.

**Comment by psychocoderHPC** [comment]
> I see `nvhpc` in the last 8th stage of the pipeline and wonder whether we should have it earlier to avoid running 7 stages to then fail in the 8th because of a very unique compiler.

I also thought about it, if so we should reorder the yaml file stay with simple indexing within the file. We could do it in a separate PR but should finish first our open PRs.

File: example/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,9 @@ add_subdirectory("kernelSpecialization/")
 add_subdirectory("monteCarloIntegration/")
 add_subdirectory("openMPSchedule/")
 add_subdirectory("randomStrategies/")
-add_subdirectory("randomCells2D/")
+# This example is producing invalid memory access during CI tests with the OpenACC backend.
+if(NOT ALPAKA_CI OR NOT ALPAKA_ACC_ANY_BT_OACC_ENABLE)
+    add_subdirectory("randomCells2D/")
+endif()
```
Comment (bernhardmgruber) [inline example/CMakeLists.txt:42]: This needs to be fixed at some point (discussed offline): #1540.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,3 +1,14 @@
+stages:
+  - stage0
+  - stage1
+  - stage2
+  - stage3
+  - stage4
+  - stage5
+  - stage6
+  - stage7
+  - stage8
```
Comment (bernhardmgruber) [inline .gitlab-ci.yml:10]: I am worried that the pipeline is too deep and CI builds take too long. But we will see.

**Review by bernhardmgruber — APPROVED** [review]

File: example/CMakeLists.txt (hunk 1 above)
Comment (psychocoderHPC) [reply example/CMakeLists.txt:42]: Thanks for opening the issue. It was to late yesterday night to do it.

**Pull request merged**